pkg/registry/apps/daemonset/storage/storage.go

```diff
@@ -108,3 +135,60 @@ func (r *StatusREST) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
 func (r *StatusREST) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
 	return r.store.ConvertToTable(ctx, object, tableOptions)
 }
+
+// ScaleREST implements a Scale for DaemonSet.
+type ScaleREST struct {
+	store *genericregistry.Store
+}
+
+// ScaleREST implements only Getter, /scale is read-only by design
+var _ = rest.Getter(&ScaleREST{})
+var _ = rest.GroupVersionKindProvider(&ScaleREST{})
+
+// GroupVersionKind returns GroupVersionKind for DaemonSet Scale object
+func (r *ScaleREST) GroupVersionKind(containingGV schema.GroupVersion) schema.GroupVersionKind {
+	return autoscalingv1.SchemeGroupVersion.WithKind("Scale")
+}
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &autoscaling.Scale{}
+}
+
+// Get retrieves object from Scale storage.
+func (r *ScaleREST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	obj, err := r.store.Get(ctx, name, options)
+	if err != nil {
+		return nil, errors.NewNotFound(apps.Resource("daemonsets/scale"), name)
+	}
+	ds := obj.(*apps.DaemonSet)
+	scale, err := scaleFromDaemonSet(ds)
+	if err != nil {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("%v", err))
+	}
+	return scale, err
+}
+
+// scaleFromDaemonSet returns a scale subresource for a DaemonSet.
+func scaleFromDaemonSet(ds *apps.DaemonSet) (*autoscaling.Scale, error) {
+	selector, err := metav1.LabelSelectorAsSelector(ds.Spec.Selector)
+	if err != nil {
+		return nil, err
+	}
+	return &autoscaling.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              ds.Name,
+			Namespace:         ds.Namespace,
+			UID:               ds.UID,
+			ResourceVersion:   ds.ResourceVersion,
+			CreationTimestamp: ds.CreationTimestamp,
+		},
+		Spec: autoscaling.ScaleSpec{
+			Replicas: ds.Status.DesiredNumberScheduled,
```

Comment:
Yes, daemon controller has its own scheduling logic but it is consistent for the controller itself.

The issues mentioned was about daemon controller not taking into account all scheduling constraints/options and ignoring particular nodes altogether.


We can see that the controller uses NodeShouldRunDaemonPod to decide which pods to create in 
https://github.com/kubernetes/kubernetes/blob/5d9d06aa4749ef9ad73f94fef0ef2e56719f9190/pkg/controller/daemon/daemon_controller.go#L782

Then it uses the same logic (NodeShouldRunDaemonPod) to compute the status and desiredNumberScheduled

https://github.com/kubernetes/kubernetes/blob/5d9d06aa4749ef9ad73f94fef0ef2e56719f9190/pkg/controller/daemon/daemon_controller.go#L1132


So semantically it should be equivalent to the replicas field.

 
We can see that in the same place we are also computing currentNumberScheduled 

https://github.com/kubernetes/kubernetes/blob/5d9d06aa4749ef9ad73f94fef0ef2e56719f9190/pkg/controller/daemon/daemon_controller.go#L1137 which is equivalent to the number of existing pods.

I think the status fields' names are a bit of misleading since the `Scheduled` part means that it was created/scheduled by the daemon controller and not neceserially recognized by the scheduler yet.